lib/brain.js

```diff
@@ -3180,7 +3143,81 @@ Brain.prototype._physicalDispense = function _physicalDispense () {
   emit('billDispenserDispensing')
 
   const notesToDispense = optimizeDispense(notes, this.billDispenser.dispenseLimit)
-  dispenseGenerator(notesToDispense, this.tx, this.config.completedTimeout)
+  const batchAmount = notesToDispense.length
+
+  const batchDispense = ({ dispensed, current, error }) => {
```

Comment:
I think we should break out long functions like this to regular JavaScript functions outside of the function. Better to have byte-sized pieces of logic.